src/main/java/dk/xam/jbang/Main.java

```diff
@@ -260,6 +260,42 @@ public Integer call() throws IOException {
 		}
 	}
 
+	static boolean checkVersions() {
+		String version = System.getProperty("java.version");
+		if (version.startsWith("1.")) {
+			version = version.substring(2, 3);
+		} else {
+			int dot = version.indexOf(".");
+			if (dot != -1) {
+				version = version.substring(0, dot);
+			}
+		}
+		int javaVersion = Integer.parseInt(version);
+		int javacVersion = 0;
+		try {
+			Process p = Runtime.getRuntime().exec("javac -version");
```

Comment:
you are not executing this the same way as the other code in jbang calls out for javac - find that and reuse the logic so you are not actually calling some *other* javac.

Reply:
Yes i am going through this part of inheritIO in process builder where javac is called.

Reply:
In jbang javac is called using the ProcessBuilder object but whenever i use inheritIO method and then start method on it the version of javac is displayed but its value is not readable using a bufferedreader.
Can you tell me what am i doing wrong?

Reply:
I mean, look at how `resolveInJavaHome()` are used to find the proper executable and handle the differences when running on windows. In your code you directly run `javac` which won't necessarily be the same that `resolveInJavaHome()` will find.

Reply:
Okay so I have to just pass `"javac -version"` string to `resolveInJavaHome()` and then execute it using `Runtime.getRuntime.exec()` to get the correct version depending upon the OS.

Reply:
When I implemented the above part even then I am required to use the split method to handle the output from the above part but using split and accessing the index 1 value lead to the exception.